src/box/lua/net_box.lua

```diff
@@ -772,7 +816,11 @@ end
 
 function remote_methods:ping(opts)
     check_remote_arg(self, 'ping')
-    return (pcall(self._request, self, M_PING, opts, nil, self._stream_id))
+    if type(opts) == 'table' and opts.is_async then
+        error("conn:ping() doesn't support `is_async` argument")
+    end
```

Comment:
I don't quite like that you access `opts` before checking it with `check_param_table`. I'd prefer to call `check_param_table` directly here. In fact, I'd consider calling `check_param_table` in each net.box method directly. The code'd become a bit more verbose, but IMO it's better to have user argument checks in the beginning of each public function instead of burying it inside an internal helper function. A potentially good thing about checking arguments at the top level is that we could set a proper [Lua error level](https://www.lua.org/pil/8.5.html) so that the user would get a file/line reference to their code (currently, this isn't applicable to `box.error` though, only to vanilla Lua `error`). Not insisting. Up to you and @Gerold103. Other than that, the patch looks good to me.

Reply:
I agree here. Accessing the opts before we properly check it doesn't look good. Apparently we need to put the opts check again outside of `_request`/`_request_impl` :)

Reply:
Yep, it looks better this way. Fixed.